src/BatchReader/IPFS.ts

```diff
@@ -48,7 +50,10 @@ export class IPFS {
 
   getDirectoryFileHashes: getDirectoryFileHashes = async (hash: string) => {
     const response = await this.ls(hash)
-    return response.Objects[hash].Links.map(x => x.Hash)
+    const validResponse = isNotNil(response.Objects) &&
```

Comment:
Dunno if needed, but maybe add isNotNil(response) just in case? or response will never be undefined or a primitive?